include/boost/context/continuation.hpp

```diff
@@ -239,15 +239,16 @@ struct result_type< Arg > {
 
 }
 
-template< typename Ctx, typename Fn, typename ... Arg >
+template< typename Ctx, typename Fn, typename Arg >
 detail::transfer_t context_ontop( detail::transfer_t t) {
-    auto p = static_cast< std::tuple< Fn, std::tuple< Arg ... > > * >( t.data);
```

Comment:
Note that the expected type for `t.data` is `std::tuple< Fn, std::tuple< Arg ... > > *`, which does not correspond to the argument type declared on line 357 below.

My guess is that the C++ library implementation for `std::tuple` changed between g++-4.9 and g++-5, which is why this bad cast happened to work with newer g++ versions but fails with version 4.9.